src/MainUtils.cpp

```diff
@@ -251,15 +251,27 @@ void genConstPackObj(const mlir::OwningModuleRef &module,
 }
 
 // Write LLVM bitcode.
-void genLLVMBitcode(const mlir::OwningModuleRef &module, string bitcodePath) {
+void genLLVMBitcode(const mlir::OwningModuleRef &module, string bitcodePath,
+    string outputBaseName) {
   error_code error;
 
+  // Write bitcode to a temporary file.
+  string bitcodeTempPath = outputBaseName + ".tmp.bc";
+  llvm::FileRemover bitcodeTempRemover(bitcodeTempPath);
+
   llvm::raw_fd_ostream moduleBitcodeStream(
-      bitcodePath, error, llvm::sys::fs::F_None);
+      bitcodeTempPath, error, llvm::sys::fs::F_None);
 
   llvm::WriteBitcodeToFile(
       *mlir::translateModuleToLLVMIR(*module), moduleBitcodeStream);
   moduleBitcodeStream.flush();
+
+  // Optimize the bitcode.
+  Command optBitcode(/*exePath=*/kOptPath);
+  optBitcode.appendStr("-O3")
+      .appendList({"-o", bitcodePath})
+      .appendStr(bitcodeTempPath)
```

Comment:
Can we use more expressive variable names? I.e., 
`bitcodePath` -> `optimizedBitCodePath`
`bitcodeTempPath` -> `unoptimizedBitCodePath`

Reply:
Tian, it's done. Thanks!